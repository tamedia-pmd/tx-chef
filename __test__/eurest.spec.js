const fn = require('../helpers/eurest');

jest.mock('axios');

test('exports a promise with menu data', () => {
  expect.assertions(2);
  return fn().then(data => {
    expect(data.date).toBe('22.09.21');
    expect(data).toMatchSnapshot();
  });
});

test('gets first meal', () => {
  expect.assertions(4);
  return fn().then(data => {
    const meal = Object.values(data.meals).filter(mealObj => mealObj.category === 'Brasserie').pop();

    expect(meal.title).toBe('Healthy Bagel');
    expect(meal.description).toBe('Kraftkorn-Malz Bagel mit Avocado, Tomaten, RÃ¼hrei und Rauchlachs (NO)');
    expect(meal.prices[0]).toBe('CHF 9,50');
    expect(meal.prices[1]).toBe('Ext. 13,50');
  });
});

