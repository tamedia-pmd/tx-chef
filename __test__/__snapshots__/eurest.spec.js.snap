// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`exports a promise with menu data 1`] = `
Object {
  "date": "22.09.21",
  "meals": Object {
    "0": Object {
      "category": "Brasserie",
      "description": "Kraftkorn-Malz Bagel mit Avocado, Tomaten, Rührei und Rauchlachs (NO)",
      "id": "0",
      "prices": Array [
        "CHF 9,50",
        "Ext. 13,50",
        "723 kcal",
      ],
      "title": "Healthy Bagel",
    },
    "1": Object {
      "category": "Grün und natürlich",
      "description": "auf Tomaten-Risotto mit Gartenkresse und Reibkäse",
      "id": "1",
      "prices": Array [
        "CHF 9,50",
        "Ext. 13,50",
        "927 kcal",
      ],
      "title": "Panierte Champignons",
    },
    "2": Object {
      "category": "Tagessuppe",
      "description": "mit Quinoa und Haselnüsse",
      "id": "2",
      "prices": Array [
        "CHF 1,40",
        "Ext. 1,70",
      ],
      "title": "Selleriesuppe",
    },
    "3": Object {
      "category": "Buddha Bowl",
      "description": "Hackfleisch(CH),Mango,Radieschen, Minze Limettenblätter, Lauchzwiebeln, Paprika, Koriander, Lattich und Sweet-Chili-Sauce",
      "id": "3",
      "prices": Array [
        "CHF 15,00",
        "Ext. 19,00",
        "679 kcal",
      ],
      "title": "Yumi Glass Nudeln",
    },
  },
}
`;
